File: DataServer/lib/mice/player.ex
defmodule DS.Mice.Player do
  alias DS.Database.User
  alias __MODULE__, as: Self

  defstruct(
    name: "",
    moid: 0,
    exp: 0,
    rank: 1,
    device_id: ""
  )

  def new(%User{}=user) do
    %Self{
      moid: user.id,
      name: user.nickname,
      device_id: "noString"
    }
  end

  def auth_response(self), do: [".auth", %{
    name: self.name,
    game: "ggc", # ggl for release
    deviceid: self.device_id,
    exp: self.exp,
    moid: self.moid,
    version: "298288", # 326539 for release
    time: 1,
    xmpp: %{
      host: "127.0.0.1"
    }
  }]

end

File: DataServer/mice/player.py

class Player:
    def __init__(self, client, user):
        self.exp = 0
        self.rank = 1
        self.client = client
        self.id = int(user['id'])
        self.name = user['nickname']
        self.device_id = 'noString'

    def auth_response(self):
        return ['.auth', {
            'time': 1,
            'moid': self.id,
            'exp': self.exp,
            'rank': self.rank,
            'name': self.name,
            'deviceid': self.device_id,
            'gameid': 'ggc', # 'ggl' on retail
            'version': '298288', # '326539' on retail
            'xmpp': { 'host': '127.0.0.1' }
        }]
